examples/python/ibmq/using_qiskit_terra_level_2.py

```diff
@@ -46,60 +37,52 @@
 
 # Making another circuit: GHZ State
 qc2 = QuantumCircuit(4, 4)
-qc2.h([0, 1, 2, 3])
+qc2.h([0])
 qc2.cx(0, 1)
 qc2.cx(0, 2)
 qc2.cx(0, 3)
 qc2.measure([0, 1, 2, 3], [0, 1, 2, 3])
 
+
+def status_header(sort):
+    print(f"{sort:<25} |  jobs | status")
+    print("===========================================")
+
+
+def format_status(status):
+    status = status.to_dict()
+    print(f"{status['backend_name'][:25]:>25} | {status['pending_jobs']:5d} | {status['status_msg'][:6]:>6}")
+
+
 # Setting up the backend
-print("(Aer Backends)")
+status_header("Aer Backends")
 for backend in BasicAer.backends():
-    print(backend.status())
+    format_status(backend.status())
 qasm_simulator = BasicAer.get_backend('qasm_simulator')
 
 
 # Compile and run the circuit on a real device backend
 # See a list of available remote backends
-print("\n(IBMQ Backends)")
+status_header("IBMQ Backends")
 for backend in provider.backends():
-    print(backend.status())
+    format_status(backend.status())
 
 try:
     # select least busy available device and execute.
-    least_busy_device = least_busy(provider.backends(simulator=False))
+    least_busy_device = least_busy([backend for backend in provider.backends(simulator=False)
+                                    if backend.status().pending_jobs > 0])
```

Comment:
This is okay as is, but `provider.backends` also supports filtering directly, e.g.
```
provider.backends(simulator=False,
                  operational=True,
                  filters=lambda b: b.configuration().n_qubits >= 5))
```